flow/layers/container_layer.h

```diff
@@ -25,22 +25,54 @@ class ContainerLayer : public Layer {
 
   const std::vector<std::shared_ptr<Layer>>& layers() const { return layers_; }
 
+  // Called when the layer, which must be a child of this container,
+  // changes its tree_reads_surface() result.
+  void update_child_readback(Layer* layer);
+
  protected:
   void PrerollChildren(PrerollContext* context,
                        const SkMatrix& child_matrix,
                        SkRect* child_paint_bounds);
   void PaintChildren(PaintContext& context) const;
 
+  virtual bool compute_tree_reads_surface() override;
+
 #if defined(OS_FUCHSIA)
   void UpdateSceneChildren(SceneUpdateContext& context);
 #endif  // defined(OS_FUCHSIA)
 
+  // Specify whether or not the container has its children render
+  // to a SaveLayer which will prevent many rendering anomalies
+  // from propagating to the parent - such as if the children
+  // read back from the surface on which they render, or if the
+  // children perform non-associative rendering. Those children
+  // will now be performing those operations on the SaveLayer
+  // rather than the layer that this container renders onto.
+  void set_renders_to_save_layer(bool protects) {
+    if (renders_to_save_layer_ != protects) {
```

Comment:
Nit: Here and later on down. Please move these out of line.

Reply:
The guidelines say inline functions should be "small, 10 lines or fewer". That would appear to match these few setter functions. They are also all basically setters with a conditional that might trigger a more extensive operation - which is out-of-line - but their main work is just to set and test a property, so it felt right to have them be inline.

Reply:
We aggressively out-of-line (and so does Chrome, they even have a Clang plugin that warns of this) to save on code size. Notice our preference for OOL-ing constructors and destructors  (if only to default them in the implementation). So where possible, please prefer moving it OOL.

Reply:
All methods more than just basically 1-line accessors should have now been OOLed.